application/common/models/User.php

```diff
@@ -426,13 +435,42 @@ public function getDisplayName()
     }
 
     /**
-     * Is it time to nag the user to add an MFA?
-     *
-     * @return bool
+     * Based on current time and presence of MFA and Method options,
+     * determine which "nag" to present to the user.
      */
-    public function isTimeToNagForMfa()
+    public function getNagState()
     {
-        return strtotime($this->nag_for_mfa_after) < time();
+        /*
+         * Don't recalculate in case the date has changed since the last calculation.
+         */
+        if ($this->nagState !== null) return $this->nagState;
+
+        $now = time();
+
+        if (strtotime($this->nag_for_mfa_after) < $now) {
```

Comment:
As a side note, [strtotime](https://php.net/strtotime) returns `false` if it fails to understand the date/time string, so if `$this->nag_for_mfa_after` isn't understandable, this condition will evaluate to true and the user will be nagged.

If that's a problem, it may be worth extracting this logic to a new method that intelligently handles invalid date/time strings somehow.

Reply:
I don't see it as a problem since `nag_for_mfa_after` is coming from the database. Will consider this potential issue in the future.

Reply:
I handled this while refactoring getNagState